### DISCUSSION TITLE
How do I remove __NUXT__ data in script tags from html pages in Nuxt.js 3？

### DISCUSSION THREAD

**Discussion by XiaoNode** [nuxt/nuxt#16122] [Questions] [answered]
After html rendering, `__NUXT__` data is too large。

when i use Nuxt.js 2, i can use hook in nuxt.config to solve it:
```javascript
  hooks: {
    "vue-renderer:ssr:context"(context) {
      const routePath = JSON.stringify(context.nuxt.routePath);
      context.nuxt = { serverRendered: true, routePath };
    },
  },
```
What should I do in Nuxt.js 3?

**Comment by TheAlexLichter** [comment, marked as answer]
Please be aware that you need `__NUXT__` for hydration and should not remove it 🙈

**Reply by tmotyl** [reply]
is there a way to prevent hydration and __nuxt_data__ even when the whole page is SSR'ed ? having the plain text __nuxt_data__ content in the html also causes seo issues, see https://github.com/nuxt/nuxt/issues/24175
